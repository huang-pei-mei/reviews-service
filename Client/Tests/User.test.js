import 'jsdom-global/register';
import React from 'react';
import { shallow, mount, render } from 'enzyme';
import "@babel/polyfill";
import User from '../src/Components/user.js';
import mockSuccessAjax from './mockSuccessAjax';
import successData  from './successData.json';
import renderer from 'react-test-renderer';

it ('should render nameAndDate if date is missing', () =>{
  const data = mockSuccessAjax(successData);
  let review = data.slice(0, 1);
  review = review[0];
  let realDate = review.date;
  review.date = undefined;
  const userBody = shallow(<User review={review} />);
  expect(userBody.find('.nameAndDate').exists()).toBe(true);
  review.date = realDate;
})

it ('should render name and date if date exists', () => {
  const data = mockSuccessAjax(successData);
  let review = data.slice(0, 1);
  review = review[0];
  const userBody = shallow(<User review={review} />);
  expect(userBody.find('.nameAndDate').exists()).toBe(true);
})

// it ('should match snapshot', () => {
//   const data = mockSuccessAjax(successData);
//   let review = data.slice(0, 1);
//   review = review[0];
//   const tree = renderer.create(<User review={review}/>).toJSON();
//   expect(tree).toMatchSnapshot();
// })