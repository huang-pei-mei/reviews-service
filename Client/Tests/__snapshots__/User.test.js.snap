// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="userInfo"
>
  <img
    alt="Profile Picture"
    className="profilePicture user"
    src="https://reviewimagesbucket.s3-us-west-1.amazonaws.com/pexels-photo-825949.jpeg"
  />
  <div
    className="nameAndDate"
  >
    <div
      className="userName"
    >
      Kathryn Russel
    </div>
    <div
      className="date"
    >
      05-14-2021
    </div>
  </div>
</div>
`;
